biolearn/data_library.py

```diff
@@ -256,6 +258,89 @@ def quality_report(self, sites=None):
 
         return QualityReport(sample_report, methylation_site_report, summary)
 
+    @staticmethod
+    def plot_sample_deviations(datasets, dataset_ids, output_path=None):
+        """
+        Creates a GeoData instance from a list of datasets and dataset ids
+        that can be used to visualize the distribution of sample deviations 
+        from the population mean in a ridge density plot.
+
+        Args:
+            datasets (list): A list of GeoData objects for each dataset.
+            dataset_ids (list): A list of dataset IDs (e.g., GSE IDs) for labeling.
+            output_path (str, optional): File path to save the plot. If None, displays the plot.
+
+        Returns:
+            None
+        """
+        # Ensure inputs are lists for uniform processing
+        if not isinstance(datasets, list):
+            datasets = [datasets]
+        if not isinstance(dataset_ids, list):
+            dataset_ids = [dataset_ids]
+
+        # Create a DataFrame to store deviations and dataset IDs
+        combined_data = []
+
+        for geodata, dataset_id in zip(datasets, dataset_ids):
+            quality_report = geodata.quality_report()
```

Comment:
How long does the quality report take to run? It might be better if this function took the quality report object as if they used had already calculated it then they would not need to calculate it again.